mopidy_tidal/heap.py

```diff
@@ -0,0 +1,50 @@
+"""A heap implementation for the lru cache."""
+from enum import Enum
+from heapq import heappop, heappush
+from itertools import count
+from typing import Any
+
+
+class Heap:
```

Comment:
Is there any reason to define a new `Heap` class here instead of using the cache mechanisms already provided by `lru_cache.py`?

Reply:
Yes: the LRU cache there isn't actually least recently *used*: it's least recently *added* (unless I've misread the code). But for playback we care about access.  I couldn't see an easy way to get what we wanted by extending `lru_cache`.

Reply:
The use case is not exactly the same indeed, but IMHO it could be covered by redefining the `__getitem__`, `__setitem__` and `__contains__` methods. The good thing is that the `LruCache` comes with a lot of the boilerplate management for persisted caches - filesystem serialize/deserialize, indexing etc.

Reply:
I'll have a look at that.  But in this case there's no need for the boilerplate--the only thing which has to be persistent is the order of entry, and we can use the os's access time for that.

That said I'll have another look at it soon.